src/webviews/extension-side/dataviewer/dataViewerDelegator.ts

```diff
@@ -14,46 +13,52 @@ import * as localize from '../../../platform/common/utils/localize';
-                    if (thirdPartyViewers.length === 1) {
-                        const command = thirdPartyViewers[0].jupyterVariableViewers.command;
-                        return commands.executeCommand(command, variable);
-                    }
-                    // show quick pick
-                    const quickPick = window.createQuickPick<QuickPickItem & { command: string }>();
-                    quickPick.title = 'Select DataFrame Viewer';
-                    quickPick.items = variableViewers.map((d) => {
-                        return {
-                            label: d.jupyterVariableViewers.title,
-                            detail: d.extension.packageJSON?.displayName ?? d.extension.id,
-                            command: d.jupyterVariableViewers.command
-                        };
-                    });
-                    quickPick.onDidAccept(async () => {
-                        const item = quickPick.selectedItems[0];
-                        if (item) {
-                            quickPick.hide();
-                            return commands.executeCommand(item.command, variable);
+            // jupyterVariableViewers
+            const variableViewers = this.getMatchingExternalVariableViewers(variable);
+            if (variableViewers.length === 0) {
+                // No data frame viewer extensions, show notifications
+                return window
+                    .showInformationMessage(
+                        localize.DataScience.dataViewerExtensionRequired,
+                        { modal: true },
+                        localize.Common.bannerLabelYes
+                    )
+                    .then((answer) => {
+                        if (answer === localize.Common.bannerLabelYes) {
+                            commands
+                                .executeCommand('workbench.extensions.search', '@tag:jupyterVariableViewers')
+                                .then(noop, noop);
                         }
                     });
-                    quickPick.show();
-                }
+            } else if (variableViewers.length === 1) {
+                const command = variableViewers[0].jupyterVariableViewers.command;
+                return commands.executeCommand(command, variable);
             } else {
-                return commands.executeCommand(Commands.ShowJupyterDataViewer, variable);
+                const thirdPartyViewers = variableViewers.filter((d) => d.extension.id !== JVSC_EXTENSION_ID);
+                if (thirdPartyViewers.length === 1) {
+                    const command = thirdPartyViewers[0].jupyterVariableViewers.command;
+                    return commands.executeCommand(command, variable);
+                }
+                // show quick pick
+                const quickPick = window.createQuickPick<QuickPickItem & { command: string }>();
+                quickPick.title = 'Select DataFrame Viewer';
```

Comment:
Needs localization, I see it's not new though

Reply:
Right, it's not new. With that said, it's better to get it localized.